dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -286,6 +286,15 @@ AppNotificationAssets Microsoft::Windows::AppNotifications::Helpers::GetAssets()
     return assets;
 }
 
+AppNotificationAssets Microsoft::Windows::AppNotifications::Helpers::GetAssets(std::wstring displayName,std::wstring iconFilePath)
+{
+    winrt::check_bool(std::filesystem::exists(iconFilePath));
+
+    THROW_HR_IF_MSG(E_INVALIDARG, !IsIconFileExtensionSupported(std::filesystem::path{ iconFilePath }), "Icon format not supported");
```

Comment:
You are implicitly creating 3 copies of the iconFilePath: one for the parameter (hstring->wstring), one for the check_bool (wstring->filesystem) and the same for verifying icon file extension. You can at least save one copy.

Also, I'm wondering if you can create a path directly from a hstring?